File: 66. Plus One/66.PlusOne.java

class Solution {
    public int[] plusOne(int[] digits) {
        // Traverse the array from the end to the beginning
        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i]++; // Increment the current digit

            // If the digit is less than 10, we can return the result
            if (digits[i] < 10) {
                return digits;
            }

            // If the digit is 10, set it to 0 and carry over the 1
            digits[i] = 0;
        }

        // If all digits were 9, add a 1 at the beginning
        int[] newDigits = new int[digits.length + 1];
        newDigits[0] = 1;

        return newDigits;
    }
}

File: 66. Plus One/66.PlusOne.js
var plusOne = function (digits) {
  // Traverse the array from the last digit to the first
  for (let i = digits.length - 1; i >= 0; i--) {
    digits[i]++; // Increment the current digit

    // If the digit becomes less than 10, no carry, return the array
    if (digits[i] < 10) {
      return digits;
    }

    // If the digit is 10, set it to 0 and carry over the 1
    digits[i] = 0;
  }

  // If we're out of the loop and all digits were 9, we need to add a new digit
  digits.unshift(1);

  return digits;
};
